lib/modules/platform/github/index.ts

```diff
@@ -1959,27 +1958,19 @@ export async function getVulnerabilityAlerts(): Promise<VulnerabilityAlert[]> {
     logger.debug('No vulnerability alerts enabled for repo');
     return [];
   }
-  let vulnerabilityAlerts: { node: VulnerabilityAlert }[] | undefined;
-
-  // TODO #22198
-  const gheSupportsStateFilter = semver.satisfies(
-    // semver not null safe, accepts null and undefined
-
-    platformConfig.gheVersion!,
-    '>=3.5',
-  );
-  const filterByState = !platformConfig.isGhe || gheSupportsStateFilter;
-  const query = vulnerabilityAlertsQuery(filterByState);
+  let vulnerabilityAlerts: VulnerabilityAlert[] | undefined;
 
   try {
-    vulnerabilityAlerts = await githubApi.queryRepoField<{
-      node: VulnerabilityAlert;
-    }>(query, 'vulnerabilityAlerts', {
-      variables: { owner: config.repositoryOwner, name: config.repositoryName },
-      paginate: false,
-      acceptHeader: 'application/vnd.github.vixen-preview+json',
-      readOnly: true,
-    });
+    vulnerabilityAlerts = (
+      await githubApi.getJson<VulnerabilityAlert[]>(
```

Comment:
we should use a zod schema 😏

Reply:
This PR has many changes already, can I add it in a follow up PR?

Reply:
The vulnerability interface has been changed enough that I consider it a rewrite --> please add directly a zod schema